### PR TITLE
[Eager] Rename EagerPyLayer to PyLayer

### PR DESCRIPTION
**Pull request by wanghuancoder**
### PR types

Others 
### PR changes

Others 
### Describe

用户端不再使用EagerPyLayer，新老动态图统一使用PyLayer，并且修复单测

### CODE REVIEW

File: python/paddle/autograd/__init__.py [hunk 1]
```diff
@@ -17,7 +17,17 @@
 from ..framework import is_grad_enabled, set_grad_enabled  # noqa: F401
 from . import backward_mode  # noqa: F401
 from .backward_mode import backward  # noqa: F401
-from .py_layer import PyLayer, PyLayerContext, EagerPyLayer, EagerPyLayerContext  # noqa: F401
+import os
+if os.environ.get('FLAGS_enable_eager_mode') == '1':
```
Comment (JiabinYang) [inline python/paddle/autograd/__init__.py]: maybe use `_in_eager_mode_` is better？

**Review by JiabinYang — COMMENTED** [review]
some comments

File: python/paddle/autograd/__init__.py (hunk 1 above)
Comment (wanghuancoder) [reply python/paddle/autograd/__init__.py]: done

File: python/paddle/distributed/fleet/meta_parallel/pp_utils/utils.py [hunk 2]
```diff
@@ -408,7 +279,7 @@ def backward(ctx, *args):
             # actually backward
             paddle.autograd.backward(forward_outputs_with_grad, backward_inputs)
             grads = tuple(inp._grad_ivar() for inp in detached_inputs
```
Comment (JiabinYang) [inline python/paddle/distributed/fleet/meta_parallel/pp_utils/utils.py:281]: maybe support list in Legacy

**Review by JiabinYang — COMMENTED** [review]
some commenst

File: python/paddle/fluid/framework.py [hunk 3]
```diff
@@ -75,7 +75,7 @@
 CONTROL_DEP_VAR_PREFIX = core.kControlDepVarName()
 
 _dygraph_tracer_ = None
-_in_eager_mode_ = (os.environ.get('FLAGS_enable_eager_mode') == '1')
+_in_eager_mode_ = (os.environ.get('FLAGS_enable_eager_mode', '1') == '1')
```
Comment (JiabinYang) [inline python/paddle/fluid/framework.py]: This change may cause eager mode enabled as default, that is not what we expected now

**Review by JiabinYang — COMMENTED** [review]
some comments

File: python/paddle/autograd/py_layer.py [hunk 4]
```diff
@@ -425,11 +425,13 @@ def mark_non_differentiable(self, *args):
         Examples:
             .. code-block:: python
 
+                import os
+                os.environ['FLAGS_enable_eager_mode'] = '1'
                 import paddle
-                from paddle.autograd import PyLayer
+                from paddle.autograd import EagerPyLayer
```
Comment (JiabinYang) [inline python/paddle/autograd/py_layer.py]: sample code should not import EagerPyLayer

File: python/paddle/autograd/py_layer.py [hunk 5]
```diff
@@ -425,11 +425,13 @@ def mark_non_differentiable(self, *args):
...
                 import numpy as np
 
-                class Tanh(PyLayer):
+                class Tanh(EagerPyLayer):
```
Comment (JiabinYang) [inline python/paddle/autograd/py_layer.py]: same

File: python/paddle/autograd/py_layer.py [hunk 6]
```diff
@@ -464,11 +466,13 @@ def set_materialize_grads(self, value: bool):
         Examples:
             .. code-block:: python
 
+                import os
+                os.environ['FLAGS_enable_eager_mode'] = '1'
                 import paddle
-                from paddle.autograd import PyLayer
+                from paddle.autograd import EagerPyLayer
```
Comment (JiabinYang) [inline python/paddle/autograd/py_layer.py]: same

File: python/paddle/autograd/py_layer.py [hunk 7]
```diff
@@ -464,11 +466,13 @@ def set_materialize_grads(self, value: bool):
...
                 import numpy as np
 
-                class Tanh(PyLayer):
+                class Tanh(EagerPyLayer):
```
Comment (JiabinYang) [inline python/paddle/autograd/py_layer.py]: same

File: python/paddle/autograd/py_layer.py [hunk 8]
```diff
@@ -478,7 +482,7 @@ def backward(ctx, grad, grad2):
                         assert np.equal(grad2.numpy(), paddle.zeros([1]).numpy())
                         return grad
 
-                class Tanh2(PyLayer):
+                class Tanh2(EagerPyLayer):
```
Comment (JiabinYang) [inline python/paddle/autograd/py_layer.py]: same

**Review by JiabinYang — COMMENTED** [review]
some comment

**Review by JiabinYang — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]
LGTM for 
`set_tests_properties(test_dygraph_sharding_stage3_for_eager PROPERTIES TIMEOUT 350)
  set_tests_properties(test_dygraph_group_sharded_api_for_eager PROPERTIES TIMEOUT 120)
`

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**